### PR TITLE
Address Issue #492

### PR DESCRIPTION
**Pull request by tomjkidd**
This fixes issue #492, handling an invalid page parameter for both the html and json type of responses.

### CODE REVIEW

File: src/clojars/web.clj [hunk 1]
```diff
@@ -39,6 +39,21 @@
...
              [session :refer [wrap-session]]]))
 
+(defn try-parse-page
+  "Will throw a targeted error if maybe-page doesn't parse as an integer."
+  [maybe-page]
+  (try
+    (let [i (Integer/parseInt maybe-page)]
+      i)
+    (catch NumberFormatException nfe
+      (let [e (ex-info
```
Comment (tobias) [inline src/clojars/web.clj]: Is there a reason you used a `let` here (and above)? If not, it would be nice to get rid of those.

Comment (tomjkidd) [inline src/clojars/web.clj]: No real reason for above, for below, it was just to organize all the ex-info args while I was writing it. I can change it easily to remove them both.

Comment (tobias) [inline src/clojars/web.clj]: Great, thanks!

**Comment by tobias** [comment]
We also return a 400 from the api here: https://github.com/clojars/clojars-web/tree/master/src/clojars/web/search.clj#L29. It would be nice to apply `error-api-response` there as well.

Would you be willing to add some tests for this as well?

**Comment by tomjkidd** [comment]
I did see that api message, just wanted to get feedback before I changed too much.
As for tests, I have no problem trying to write some to cover the cases. I ran into issues using `lein test` though, because I am on Windows, so it'll be flying blind to some extent (notes from my experience in the issue)

**Comment by tomjkidd** [comment]
I was able to figure out what was causing migrations to crash, so I also updated that to no be OS specific. I've also now added the tests. Thanks for the feedback.

**Comment by tobias** [comment]
Merged, thanks!

**Pull request closed without merging**